src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala

```diff
@@ -318,9 +322,15 @@ final case class ClientsApiServiceImpl(
     key: AuthorizationProcessModel.Key
   )(implicit contexts: Seq[(String, String)]): Future[PublicKey] = {
     for {
-      selfcareUuid <- getSelfcareIdFutureUUID(contexts)
-      user         <- selfcareV2ClientService.getUserById(selfcareUuid, key.userId).map(_.toApi).flatMap(_.toFuture)
-    } yield key.toApi(user)
+      selfcareUuid     <- getSelfcareIdFutureUUID(contexts)
+      userResponse     <- selfcareV2ClientService.getUserById(selfcareUuid, key.userId).recoverWith {
+        case _: UserNotFound => Future.successful(UserResponse())
+      }
+      (user, isOrphan) <- userResponse match {
+        case UserResponse(_, id, _, _, _) if id == None => userResponse.toApi.toFuture.zip(Future.successful(true))
+        case _                                          => userResponse.toApi.toFuture.zip(Future.successful(false))
+      }
```

Comment:
`zip` leverages on execution context, which is not needed here (no parallelism)
```suggestion
      isOrphan = userResponse.id.isEmpty
      user <- userResponse.toApi.toFuture
```

Btw are you sure `userResponse.toApi` is ok?
The requirement here is to not fail if the user is not found, but to use placeholders.
Wouldn't `toApi` fail if there are missing fields?